railties/lib/rails/application/bootstrap.rb

```diff
@@ -39,7 +39,11 @@ module Bootstrap
           f.binmode
           f.sync = config.autoflush_log # if true make sure every write flushes
 
-          logger = ActiveSupport::Logger.new f
+          logger_args = [f]
+          logger_args << config.logger_shift_age
+          logger_args << config.logger_shift_size if config.logger_shift_age.to_i > 0
+
+          logger = ActiveSupport::Logger.new(*logger_args.compact)
```

Comment:
Yes, I just checked – we don't need to compact arguments here, because when ruby logger receives nil as concrete argument – it uses default value for that argument.